sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -130,4 +161,16 @@ private String interceptAuthorizationUri(String authorizationUri) {
         }
         return null;
     }
+
+    void replyForbiddenWithWwwAuthenticateHeader(HttpServletResponse response,
+                                                 MsalInteractionRequiredException exception) {
+        Map<String, Object> parameters = new LinkedHashMap<>();
+        response.setStatus(HttpStatus.FORBIDDEN.value());
+        parameters.put(Constants.CONDITIONAL_ACCESS_POLICY_CLAIMS,
+            exception.claims());
+        parameters.put(OAuth2ParameterNames.ERROR, OAuth2ErrorCodes.INVALID_TOKEN);
+        parameters.put(OAuth2ParameterNames.ERROR_DESCRIPTION, "The resource server requires higher privileges than "
+            + "provided by the access token");
```

Comment:
Does `MsalInteractionRequiredException ` have error message? Maybe we can use the error message instead.

Reply:
I have tired this, but we need to put the error message in the `response header`, the error message of `MsalInteractionRequiredException` will fail in [addHeader(String name, String value)](https://github.com/spring-projects/spring-security/blob/master/web/src/main/java/org/springframework/security/web/firewall/FirewalledResponse.java#L58) API.
I analyze the error message of `MsalInteractionRequiredException` and I think there's no useful information, so I customize a message as a description.